packages/icon-button/src/IconButton.tsx

```diff
@@ -240,6 +240,21 @@ function usesLinkElement(
   return props.href != null;
 }
 
+function isComponentGlyph<T extends React.ReactElement>(
+  child: React.ReactNode,
+): child is T {
```

Comment:
This is indicating to typescript that the child is simply a ReactElement, not a Glyph. Not sure if there is a more strict way to type this. Perhaps something roughly like:
```ts
function isComponentGlyph<T extends React.ReactElement>(child: React.ReactNode): child is React.ComponentType<GlyphProps> {...}
```

Reply:
👍

Reply:
If we bring in glyph props then we have to bring the entire icon library in and i'm not sure we want to do that, right?